migrator/migrations/m_171_to_m_172_move_scope_to_collection_in_report_configurations/migration.go

```diff
@@ -0,0 +1,245 @@
+			},
+		})
+		collectionsToEmbed = append(collectionsToEmbed, col)
+		index = index + 1
+	}
+
+	if includedNamespaces := scope.GetRules().GetIncludedNamespaces(); len(includedNamespaces) > 0 {
+		for _, namespace := range includedNamespaces {
+			col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+				{
+					FieldName: search.Cluster.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetClusterName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+				{
+					FieldName: search.Namespace.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetNamespaceName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+			})
+			collectionsToEmbed = append(collectionsToEmbed, col)
+			index = index + 1
+		}
+	}
+
+	if clusterLabelSelectors := scope.GetRules().GetClusterLabelSelectors(); len(clusterLabelSelectors) > 0 {
+		labelCollections, err := labelSelectorsToCollections(scope.GetName(), index, clusterLabelSelectors, search.ClusterLabel.String())
+		if err != nil {
+			return nil, err
+		}
+		index = index + len(labelCollections)
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+
+	if namespaceLabelSelectors := scope.GetRules().GetNamespaceLabelSelectors(); len(namespaceLabelSelectors) > 0 {
+		labelCollections, err := labelSelectorsToCollections(scope.GetName(), index, namespaceLabelSelectors, search.NamespaceLabel.String())
+		if err != nil {
+			return nil, err
+		}
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+	return collectionsToEmbed, nil
+}
+
+// Creates embedded and root collections for the access scope with ID scopeID.
+// Adds the embedded and root collections to the collection store.
+func createCollectionsForScope(ctx context.Context, scopeID string,
+	accessScopeStore accessScopePostgres.Store, collectionStore collectionPostgres.Store) error {
+	scope, found, err := accessScopeStore.Get(ctx, scopeID)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to fetch scope with id %s. "+scopeAttachedToConfigsTemplate,
+			scopeID, getJoinedConfigsForScopeID(scopeID))
+	}
+	if !found {
+		log.Errorf("Scope with id %s not found. "+scopeAttachedToConfigsTemplate+manualResolutionTemplate,
+			scopeID, strings.Join(scopeIDToConfigNames[scopeID], ", "))
+		return nil
+	}
+
+	collectionsToEmbed, err := getCollectionsToEmbed(scope)
+	if err != nil {
+		if strings.Contains(err.Error(), "Unsupported operator") {
+			log.Errorf("Failed to create collections for scope <%s>; Reason : %s; "+
+				scopeAttachedToConfigsTemplate+manualResolutionTemplate,
+				scope.GetName(), err.Error(), getJoinedConfigsForScopeID(scopeID))
+			return nil
+		}
+		return err
+	}
+	err = collectionStore.UpsertMany(ctx, collectionsToEmbed)
+	if err != nil {
+		return err
```

Comment:
I was erroring out in places where error comes from a postgres store because those would indicate something wrong with the store, and if one iteration errors out, there is a chance that others might too. That said, logging and moving on is a fine approach to me as well since it tries it migrate as much as possible without short circuiting on error.